Classes/common/touchdb/TD_Database.m

```diff
@@ -438,24 +577,43 @@ - (BOOL)close
     _activeReplicators = nil;
 
     [_fmdbQueue close];
+    _fmdbQueue = nil;
+    _keyProviderToOpenDB = nil;
 
     _open = NO;
     _transactionLevel = 0;
+    
     return YES;
 }
 
-- (BOOL)deleteDatabase:(NSError**)outError
+- (BOOL)deleteDatabase:(NSError **)outError
 {
-    CDTLogInfo(CDTDATASTORE_LOG_CONTEXT, @"Deleting %@", _path);
     [[NSNotificationCenter defaultCenter] postNotificationName:TD_DatabaseWillBeDeletedNotification
                                                         object:self];
-    if (_open) {
-        if (![self close]) return NO;
-    } else if (!self.exists) {
-        return YES;
+
+    if (_open && ![self close]) {
```

Comment:
Is it possible here for `_open` to be `YES`, then this thread is interrupted and the database is closed, in which case `[self close]` will return `NO`? If that can happen, we end up failing this method even though the database is closed and we could go on to delete it.

Even if the above is the case, it may not be important enough to fix, given the method could be called again fairly simply -- particularly if you fill in the `outError` appropriately.